src/ResourceManager/Kusto/Commands.Kusto/Commands/UpdateAzureRmKustoCluster.cs

```diff
@@ -0,0 +1,152 @@
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = CmdletParametersSet,
+            Mandatory = true,
+            HelpMessage = "Azure region where the cluster exists.")]
+        [LocationCompleter("Microsoft.Kusto/clusters")]
+        public string Location { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Name of the Sku used to create the cluster")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("KC8", "KC16", "KS8", "KS16", "L8", "L16", "D14_v2", "D13_v2", IgnoreCase = true)]
+        public string SkuName { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Name of the Tier used to create the cluster")]
+        [ValidateSet("Standard", IgnoreCase = true)]
+        public string Tier { get; set; }
+        
+        [Parameter(
+            ParameterSetName = ResourceIdParameterSet,
+            Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Kusto cluster ResourceID.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ObjectParameterSet,
+            Mandatory = true,
+            Position = 2,
+            ValueFromPipeline = true,
+            HelpMessage = "Kusto cluster object.")]
+        [ValidateNotNullOrEmpty]
+        public PSKustoCluster InputObject { get; set; }
+
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        public override void ExecuteCmdlet()
+        {
+            string clusterName = Name;
+            string resourceGroupName = ResourceGroupName;
+            string location = Location;
+
+            if (!string.IsNullOrEmpty(ResourceId))
+            {
+                KustoUtils.GetResourceGroupNameAndClusterNameFromClusterId(ResourceId, out resourceGroupName, out clusterName);
+            }
+            else if (InputObject != null)
+            {
+                KustoUtils.GetResourceGroupNameAndClusterNameFromClusterId(InputObject.Id, out resourceGroupName, out clusterName);
+            }
+
+            if (string.IsNullOrEmpty(clusterName))
+            {
+                WriteExceptionError(new PSArgumentNullException("Name", "Name of cluster not specified"));
+            }
+
+            if (ShouldProcess(clusterName, Resources.UpdatingKustoCluster))
+            {
+                try
+                {
+                    var cluster = KustoClient.GetCluster(resourceGroupName, clusterName);
+                    if (cluster == null)
+                    {
+                        throw new CloudException(string.Format(Resources.KustoClusterNotExist, Name));
+                    }
+
+                    if (location != null && location != cluster.Location)
```

Comment:
Can you have two cluster with the same name in the same resource group? Otherwise, the location parameter shouldn't be necessary.

Reply:
I removed the location, I also need to change the design